### ISSUE TITLE
Crash when dragging item with transition

### ISSUE THREAD

**Issue by mdmayfield** [olive-editor/olive#596]
Steps to reproduce:

1) New project, new sequence
2) Insert Solid Color item
3) Add cross dissolve transition to item that lasts its entire length
4) Switch to Move tool and drag item

Windows backtrace:
```
Thread 1 received signal SIGSEGV, Segmentation fault.
0x0000000000444b12 in Media::get_type (this=0x0) at project/media.cpp:210
210       return type;
(gdb) bt
#0  0x0000000000444b12 in Media::get_type (this=0x0) at project/media.cpp:210
#1  0x000000000043a976 in TimelineWidget::update_ghosts (this=0x4340c350,
    mouse_pos=..., lock_frame=false) at ui/timelinewidget.cpp:1800
#2  0x000000000043c240 in TimelineWidget::mouseMoveEvent (this=0x4340c350,
    event=0x76db9d0) at ui/timelinewidget.cpp:2147
#3  0x000000001b4ca3b4 in QWidget::event (this=0x4340c350, event=0x76db9d0)
    at kernel/qwidget.cpp:8880
#4  0x000000001b4882f1 in QApplicationPrivate::notify_helper (
    this=this@entry=0x35878060, receiver=receiver@entry=0x4340c350,
    e=0x76db9d0) at kernel/qapplication.cpp:3752
#5  0x000000001b48fd92 in QApplication::notify (this=<optimized out>,
    receiver=0x4340c350, e=0x76db9d0) at kernel/qapplication.cpp:3213
#6  0x000000006941a202 in QCoreApplication::notifyInternal2 (
    receiver=0x4340c350, event=0x76db9d0) at kernel/qcoreapplication.cpp:1061
#7  0x000000006941a448 in QCoreApplication::sendSpontaneousEvent (
    receiver=receiver@entry=0x4340c350, event=event@entry=0x76db9d0)
    at kernel/qcoreapplication.cpp:1463
#8  0x000000001b48ece7 in QApplicationPrivate::sendMouseEvent (
    receiver=receiver@entry=0x4340c350, event=event@entry=0x76db9d0,
    alienWidget=alienWidget@entry=0x4340c350, nativeWidget=0x76dfa20,
    buttonDown=buttonDown@entry=0x1b9d82c0 <qt_button_down>,
    lastMouseReceiver=..., spontaneous=spontaneous@entry=true,
    onlyDispatchEnterLeave=onlyDispatchEnterLeave@entry=false)
    at kernel/qapplication.cpp:2699
#9  0x000000001b4df79e in QWidgetWindow::handleMouseEvent (
    this=this@entry=0x435308a0, event=event@entry=0x76dc140)
    at C:/building/msys32/mingw64/include/c++/8.2.1/bits/atomic_base.h:390
#10 0x000000001b4e3044 in QWidgetWindow::event (this=0x435308a0,
    event=0x76dc140) at kernel/qwidgetwindow.cpp:281
#11 0x000000001b4882f1 in QApplicationPrivate::notify_helper (
    this=this@entry=0x35878060, receiver=receiver@entry=0x435308a0,
    e=e@entry=0x76dc140) at kernel/qapplication.cpp:3752
#12 0x000000001b48faa4 in QApplication::notify (this=0x76dfd90,
    receiver=0x435308a0, e=0x76dc140) at kernel/qapplication.cpp:3109
#13 0x000000006941a202 in QCoreApplication::notifyInternal2 (
    receiver=0x435308a0, event=0x76dc140) at kernel/qcoreapplication.cpp:1061
#14 0x000000006941a448 in QCoreApplication::sendSpontaneousEvent (
    receiver=receiver@entry=0x435308a0, event=event@entry=0x76dc140)
    at kernel/qcoreapplication.cpp:1463
#15 0x0000000007ea9266 in QGuiApplicationPrivate::processMouseEvent (
    e=0x43628670) at kernel/qguiapplication.cpp:2102
#16 0x0000000007eaa367 in QGuiApplicationPrivate::processWindowSystemEvent (
    e=0x0, e@entry=0x43628670) at kernel/qguiapplication.cpp:1837
#17 0x0000000007e88f2b in QWindowSystemInterface::sendWindowSystemEvents (
    flags=...) at kernel/qwindowsysteminterface.cpp:1068
#18 0x0000000035e67989 in QWindowsGuiEventDispatcher::sendPostedEvents (this=
    0x358783b0) at qwindowsguieventdispatcher.cpp:82
#19 0x000000006946ff8f in qt_internal_proc (hwnd=0x120b76, message=1025,
    wp=0, lp=0) at kernel/qeventdispatcher_win.cpp:245
#20 0x00007ffae5cfca66 in USER32!DispatchMessageW ()
   from C:\WINDOWS\System32\user32.dll
#21 0x00007ffae5cfc582 in USER32!DispatchMessageW ()
   from C:\WINDOWS\System32\user32.dll
#22 0x000000006946f75f in QEventDispatcherWin32::processEvents (
    this=0x358783b0, flags=...) at kernel/qeventdispatcher_win.cpp:639
#23 0x0000000035e67967 in QWindowsGuiEventDispatcher::processEvents (
    this=0x358783b0, flags=...) at qwindowsguieventdispatcher.cpp:74
#24 0x00000000694188d7 in QEventLoop::processEvents (
    this=this@entry=0x76df960, flags=..., flags@entry=...)
    at C:/building/msys32/mingw64/include/c++/8.2.1/bits/atomic_base.h:707
#25 0x0000000069418d07 in QEventLoop::exec (this=this@entry=0x76df960,
    flags=flags@entry=...)
    at ../../include/QtCore/../../src/corelib/global/qflags.h:122
#26 0x0000000069421ba2 in QCoreApplication::exec ()
    at ../../include/QtCore/../../src/corelib/global/qflags.h:120
#27 0x0000000007ea059f in QGuiApplication::exec ()
    at kernel/qguiapplication.cpp:1779
#28 0x000000001b48825e in QApplication::exec ()
    at kernel/qapplication.cpp:2909
#29 0x0000000000401d34 in main (argc=1, argv=0x3586cc80) at main.cpp:149
```

Linux backtrace:
```
Thread 1 "olive-editor" received signal SIGSEGV, Segmentation fault.
0x00005555555e2720 in Media::get_type() ()
(gdb) bt
#0  0x00005555555e2720 in Media::get_type() ()
#1  0x00005555555dc6bd in TimelineWidget::update_ghosts(QPoint const&, bool) ()
#2  0x00005555555ddb3c in TimelineWidget::mouseMoveEvent(QMouseEvent*) ()
#3  0x00007ffff5c667d8 in QWidget::event(QEvent*) () at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#4  0x00007ffff5c284a1 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#5  0x00007ffff5c2fd28 in QApplication::notify(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#6  0x00007ffff5269499 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#7  0x00007ffff5c2f029 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#8  0x00007ffff5c81304 in  () at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#9  0x00007ffff5c83e8e in  () at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#10 0x00007ffff5c284a1 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#11 0x00007ffff5c2fae0 in QApplication::notify(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#12 0x00007ffff5269499 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#13 0x00007ffff566e53b in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () at /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5
#14 0x00007ffff5670435 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) () at /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5
#15 0x00007ffff564ab6b in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5
#16 0x00007fffe2e48e5b in  () at /usr/lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#17 0x00007ffff526816b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#18 0x00007ffff52702e2 in QCoreApplication::exec() () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#19 0x00005555555a841a in main ()
```

**Comment by mdmayfield** [comment]
Sample project to trigger crash (simply move the one item)

[update-ghosts-crash.ove.zip](https://github.com/olive-editor/olive/files/2928012/update-ghosts-crash.ove.zip)

**Comment by itsmattkc** [comment]
Good pickup, fixed in 27d77abdc2fbc017aafe5a4c0292ddfa1e7962ee
